### ISSUE TITLE
Croppa doesn't upload new image, only initial image

### ISSUE THREAD

**Issue by joshsmith01** [zhanziyang/vue-croppa#145]
I'm loading in an initial image, but then providing users the option to change that image. If they do, I want to immediately upload that new image from Croppa to my server. The server will generate the necessary data and send data back to me. I almost have it all working. I can load in the initial image, change the image, and upload. But the uploaded image is always the initial image. 

How can I ensure that my new image is what is being sent to the server and also replace my initial image? 

Croppa component:
```javascript
<croppa
  v-show="uploadedAvatar"
  v-model="myCroppa"
  @new-image-drawn="uploadAvatar"
  @file-choose="changeImage(input, index)"
  :width="535"
  :height="449"
  :prevent-white-space="true"
  :show-remove-button="false"
  :quality="5"
  canvas-color="transparent">
  <img crossorigin="anonymous" slot="initial" :src="input.image_url" />
</croppa>
```

Method:
```javascript
async changeImage(input, index) {
        console.log('new image chosen')
        /* Upload Story Swear Image */
        input.image_url = ''
        this.myCroppa.refresh();
        this.myCroppa.generateBlob((blob) => {

          // generate random filename
          let randomFilename = Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
          let fd = new FormData()
          fd.append('file', blob, randomFilename + '.jpg')
          const uploadedImageId = this.$store.dispatch('uploadImage', fd);
          uploadedImageId
            .then((res) => {
              input.new_image_id = res
              console.log(res)
            });

          console.log(input.new_image_id)
        }, 'image/jpeg', 0.8) // 80% compressed

      }
```

**Comment by zhanziyang** [comment]
What do you do in the `uploadAvatar` method? I think you should do the upload in that method.

**Comment by zhanziyang** [comment]
On vue-croppa 2.x: Use `@new-image` or `@file-choose` to detect user selection, and export only after `@new-image-drawn` fires.

Closing as answered. Please feel free to open a new issue if this doesn't cover your use case.
